### ISSUE TITLE
Proposal  - dim images with opacity

### ISSUE THREAD

**Issue by cliffordfajardo** [darkreader/darkreader#1746]
# What is the Proposal?
Add dimming to images.

# Why?
When browsing the web at night, often times you'll run into images that have white or very bright backgrounds and are not png's. Imagine browsing websites (example e-commerce/shopping) where they use large white backgrounds.

![ImageJoiner-2019-11-17 at 12 37 06 PM](https://user-images.githubusercontent.com/6743796/69013631-2d889b00-0937-11ea-81dc-bf3bf3f52832.jpg)

# Solution
- Dim images slightly.
- Add a configurable option to toggle this feature on/off.

![demo](https://user-images.githubusercontent.com/6743796/69013486-658ede80-0935-11ea-9853-37567fb55f6b.jpg)

# Potential Challenges
- jpg's or images that don't have bright backgrounds could become hard to read.
I haven't dug into how to solve that issue too much, but I know there are micro javascript libraries that can detect the color of images and analyze the brightness level, if we were to use something like that, we could check the brightness of the image after loading the page, to undo dimming, if the image doesn't need to be dimmed.

[CSS filters have strong support already](https://caniuse.com/#feat=css-filters). Most users who use dark reader, we can assume are on relatively newer browsers. If they know how to install an extension and find one like dark reader, we can make some assumptions =).

```css
img {
	border-color: #000;
	box-shadow: none;
	-webkit-filter: brightness(0.8) contrast(1.2);
	filter: brightness(0.8) contrast(1.2);
}
```

Live demo code (JSbin.com): https://jsbin.com/naseresade/4/edit?html,output

**Comment by asd-collab** [comment]
> If they know how to install an extension and find one like dark reader

its an extension recommended by Firefox.

---

let site owners fix that bug. this is a new way to waste CPU cycles and other resources.


---

asd-collab

**Comment by WebMechanic** [comment]
> this is a new way to waste CPU cycles and other resources.

so are browser extensions ;)

Actually CSS filters run on the GPU.

**Comment by cliffordfajardo** [comment]
I haven't had a chance to look at the codebase recently, but this is still something that's still worth looking into. The following: 

> let site owners fix that bug. 

the idea runs contrary to the whole concept of web extensions in this context. Also that will likely never happen. That's like saying: _let each site owner implement browsing it in dark mode._

The reason we create chrome extension's to augment our experiences =)

**Comment by WebMechanic** [comment]
I actually like your proposal - in general, and yes: it'd need a toggle. 
It may also require a selection of **blend modes** (besides filters) that are actually designed to specifically reduce/affect whites and light colours while keeping others (mainly) intact. There'll be only a few to pick from.
- https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode
- https://caniuse.com/#search=mix-blend-mode

With images like the ones in your post, that depict a pretty clear subject against a solid colour, it'd be relatively easy to even create decent masks "on the fly" and literally cut out the main subject or use them to control filters and blend modes. 
Using Canvas2D or SVG can help as intermediates. Modern CSS combined with SVG filters can become a very powerful "image editor". The math for his is well known and JS libraries exist to do this.

Cheers.

**Comment by vyach-vasiliev** [comment]
Yes! Indeed, this stunning extension has saved my eyes more than once by nights!
It would be great to have this functionality, albeit as an implicit setting.

In the meantime, I am using such a simple code that I pasted into the `config editor`.

```css
img, video {
    filter: brightness(0.5);
}
img:hover, video:hover {
    filter: unset;
}
```
Hovering over an element returns the brightness so that you can better understand the context 👍

**Comment by Sebastian-Nielsen** [comment]
> Yes! Indeed, this stunning extension has saved my eyes more than once by nights! It would be great to have this functionality, albeit as an implicit setting.
> 
> In the meantime, I am using such a simple code that I pasted into the `config editor`.
> 
> ```css
> img, video {
>     filter: brightness(0.5);
> }
> img:hover, video:hover {
>     filter: unset;
> }
> ```
> 
> Hovering over an element returns the brightness so that you can better understand the context 👍

What do you mean by the `config editor`/where do I find it?

**Comment by rzhb** [comment]
> What do you mean by the `config editor`/where do I find it?

I think it's Darker Reader DEVELOPER TOOLS, put it under the `CSS` line.
